frontend/packages/topology/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -93,11 +94,14 @@ const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
 
   const onEnter = React.useCallback(
     (e) => {
-      if (selectedItem) {
+      const { id } = document.activeElement;
+      const activeViewAllLink = viewAll?.find((link) => link.catalogType === id);
+      if (activeViewAllLink) history.push(activeViewAllLink.to);
+      else if (selectedItem) {
         handleCta(e, selectedItem, closeModal, fireTelemetryEvent);
       }
```

Comment:
Makes it more readable in my opinion.
```suggestion
      if (activeViewAllLink) {
        history.push(activeViewAllLink.to);
      } else if (selectedItem) {
        handleCta(e, selectedItem, closeModal, fireTelemetryEvent);
      }
```